### PR TITLE
[web] Access engine version to get correct gstatic URL

### PR DESCRIPTION
**Pull request by hterkelsen**
Creates `dart:_web_sdk_version` which exposes the engine SHA so we can get the correct URL for the CanvasKit artifacts.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: third_party/web_sdk_version/lib/web_sdk_version/web_sdk_version.dart [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// NOTE: This file is just a stub. The actual file is generated at build time
+// with the actual current Git commit SHA. This stub is just here for local
+// development.
+
+/// The Git SHA of the Flutter Engine.
+const String flutterWebSdkVersion = 'IF YOU SEE THIS FILE A BUG: https://github.com/flutter/flutter/issues/new?assignees=&labels=&template=2_bug.md&title=';
```
Comment (kevmoo) [inline third_party/web_sdk_version/lib/web_sdk_version/web_sdk_version.dart:10]:
```suggestion
const String flutterWebSdkVersion = 'IF YOU SEE THIS, FILE A BUG: https://github.com/flutter/flutter/issues/new?assignees=&labels=&template=2_bug.md&title=';
```

Comment (hterkelsen) [reply third_party/web_sdk_version/lib/web_sdk_version/web_sdk_version.dart:10]: done

File: third_party/web_sdk_version/pubspec.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+name: web_sdk_version
+
+publish_to: none
+
+environment:
+  sdk: ">=2.12.0-0 <3.0.0"
```
Comment (kevmoo) [inline third_party/web_sdk_version/pubspec.yaml:6]:
```suggestion
  sdk: ">=3.0.0-0 <3.0.0"
```

might as well...

**Review by eyebrowsoffire — APPROVED** [review]
LGTM!

**Review by mdebbar — APPROVED** [review]

**Pull request merged**